test/src/test.c

```diff
@@ -173,13 +173,19 @@ p_test_impl(bool do_malloc_init, bool do_reentrant, test_t *t, va_list ap) {
 		}
 	}
 
+	if (test_counts[test_status_fail] != 0) {
+		malloc_write("\033[1;31m");
+	}
 	malloc_printf("--- %s: %u/%u, %s: %u/%u, %s: %u/%u ---\n",
```

Comment:
Sorry I didn't think of this during review earlier, but come to think of it we should probably do all of this in a single write if possible to avoid interleaving with other output (when tests are run in parallel) and ending up with garbled colors. So maybe something alone the lines of:
```c
bool test_failed = test_counts[test_status_fail] != 0;
const char *color_start = test_failed ? "\033[1;31m" : "";
const char *color_end = test_failed ? "\033[0m" : "";
```

And add those in as `%s` at the beginning and end of the existing `malloc_printf`.

Reply:
Hmm did not think of the racy conditions before. `malloc_printf` itself is not atomic, so printing them in the same line may still theoretically possible lead to unexpected results? Yet the style is indeed more aligned and looks better so modified.